Projects/Dotmim.Sync.Core/Orchestrators/ApplyChanges/BaseOrchestrator.ApplyChanges.cs

```diff
@@ -648,96 +648,96 @@ public abstract partial class BaseOrchestrator
             if ((conflictRows != null && conflictRows.Count > 0) || (errorsRows != null && errorsRows.Count > 0))
             {
 
-                await using var runnerError = await this.GetConnectionAsync(context, Options.TransactionMode == TransactionMode.None ? SyncMode.NoTransaction : SyncMode.WithTransaction, SyncStage.ChangesApplying, connection, transaction, cancellationToken, progress).ConfigureAwait(false);
-
-                // Disable check constraints for provider supporting only at table level
-                if (this.Options.DisableConstraintsOnApplyChanges && this.Provider.ConstraintsLevelAction == ConstraintsLevelAction.OnTableLevel)
-                    await this.InternalDisableConstraintsAsync(scopeInfo, context, schemaChangesTable, runnerError.Connection, runnerError.Transaction, runnerError.CancellationToken, runnerError.Progress).ConfigureAwait(false);
-
-                // If conflicts occured
-                foreach (var conflictRow in conflictRows)
+                using var runnerError = await this.GetConnectionAsync(context, Options.TransactionMode == TransactionMode.None ? SyncMode.NoTransaction : SyncMode.WithTransaction, SyncStage.ChangesApplying, connection, transaction, cancellationToken, progress).ConfigureAwait(false);
```

Comment:
the using statement on this line should be removed, since `await using` is done on the next line.

Reply:
I don't get it.
This is what I do everywhere:

``` cs
using var runner = await this.GetConnectionAsync(context, SyncMode.NoTransaction, SyncStage.ChangesSelecting, connection, transaction).ConfigureAwait(false);
await using (runner.ConfigureAwait(false))
{
  ....
}

```

If I remove the first using, I get a warning from InferSharp

![image](https://github.com/Mimetis/Dotmim.Sync/assets/4592555/af656d7f-d5db-4018-86a4-aaae57185d26)

Reply:
The latest InferSharp 1.5 release last year was meant to address false positives with async code, but `IAsyncDisposable` may still cause false positives, in my opinion.

The tool was useful for finding many real leaks, and indeed all of the currently flagged ones for master branch are real.

But I think that it isn't perfect.

For instance, the following code will also get flagged by InferSharp:

```csharp
DbConnectionRunner runner = null;

try
{
    runner = await this.FooAsync().ConfigureAwait(false);
    // ...
}
finally
{
    runner?.DisposeAsync().ConfigureAwait(false);
}
```

Reply:
If we accidentally capture the synchronization context at any point, we lose all the benefits of that.

I will probably open some PRs after this one is done to use ConfigureAwaitChecker and enforce CA2007

Reply:
I've enforced the `CA2007` rule in the `editorconfig`, and I have disabled it in the tests / sample project.
And I've fixed all the ones I forgot.
If tests are passing, I'll merge this PR

Thanks @symbiogenesis for all the good advices on this particular area !